pkg/postgres/postgres.go

```diff
@@ -25,7 +26,7 @@ type ConnectionString struct {
 
 // Raw returns a PostgreSQL connection string.
 func (c ConnectionString) Raw() string {
-	raw := fmt.Sprintf("postgresql://%s:%s@%s", c.User, c.Password, c.Host)
+	raw := fmt.Sprintf("postgresql://%s:%s@%s", c.User, url.QueryEscape(c.Password), c.Host)
```

Comment:
Can we cut this change into a separate PR? It is not related to the PostgreSQLHostCredentials change and also is a notable change in behaviour I would expect so I think it's due with its own entry in a changelog.